back-end/Service.py

```diff
@@ -0,0 +1,280 @@
+    db = DataAccessImage()
+    image_handler = ImageHandler(db)
+    image =  Image(data["name"], data["imageType"], data["eventType"], data["location"])
+    return jsonify(image_handler.edit_image(image))
+ 
+
+@app.route('/delete', methods=['POST'])
+def delete_image( ):
+    data = json.loads(request.stream.read())
+    db = DataAccessImage()
+    image_handler = ImageHandler(db)
+    image = Image(data["name"],data["imageType"],data["eventType"],data["location"])
+    return jsonify(image_handler.delete_image(image)) #RECHECK 
+
+
+@app.route('/addTimeUnit', methods=['POST'])
+def add_time_to_calendar():
+    data = json.loads(request.stream.read())
+    db = DataAccessCalendar()
+    tu_handler = TimeUnitHandler(db)
+    time_unit = TimeUnit(data["date"],data["dayOfWeek"],data["time"],data["isWeekly"] )
+    return jsonify(tu_handler.add_time_to_calendar(time_unit))
+
+@app.route('/deleteTimeUnit', methods=['POST'])
+def delete_time_from_calendar():
+    data = json.loads(request.stream.read())
+    db = DataAccessCalendar()
+    tu_handler = TimeUnitHandler(db)
+    timeunit = TimeUnit(data["date"],data["dayOfWeek"],data["time"],data["isWeekly"] )
+    return jsonify(tu_handler.delete_time_from_calendar(timeunit))
+
+@app.route('/deleteOrder', methods=['POST'])
+def delete_order():
+    data = json.loads(request.stream.read())
+    db = DataAccessOrders()
+    or_handler = OrderHandler(db)
+    return jsonify(or_handler.delete_order(data))
+             
+@app.route('/deleteLocationType', methods=['POST'])
+def delete_location_type():
+    data = json.loads(request.stream.read())
+    db = DataAccessLocation()
+    loc_handler = LocationHandler(db)
+    return jsonify(loc_handler.delete_location_type(data))
+
+
+@app.route('/updateOrder', methods=['POST'])
+def update_order():
+    data = json.loads(request.stream.read())
+    db = DataAccessOrders()
+    or_handler = OrderHandler(db)
+    order = Order( data["date"], data["time"],data["name"],data["telefon"],data["email"],data["location"], data["eventType"],data["note"], data["id"])
+    return jsonify(or_handler.update_order(order))
+
+@app.route('/getTimeSlots') 
+def get_all_time_slots():
+    db = DataAccessCalendar()
+    tu_handler = TimeUnitHandler(db)
+    return jsonify(tu_handler.get_time_slots())
+
+
+@app.route('/getWeeklyOpenSlots')
+def get_weekly_slots():
+    db = DataAccessCalendar()
+    tu_handler = TimeUnitHandler(db)
+    return jsonify(tu_handler.get_weekly_time_slots())
+
+@app.route('/getSingleOpenSlots')
+def get_single_slots():
+    db = DataAccessCalendar()
+    tu_handler = TimeUnitHandler(db)
+    return jsonify(tu_handler.get_single_time_slots())
+
+
+@app.route('/addorder', methods=['POST']) 
+def add_order():
+    data = json.loads(request.stream.read())
+    db = DataAccessOrders()
+    or_handler = OrderHandler(db)
+    order = Order( data["date"], data["time"],data["name"],data["telefon"],data["email"],data["location"], data["eventType"],data["note"])

```

Comment:
consider doing validation for the request payloads somehow

Reply:
Do you have some articles for that?